src/js/components/Button/StyledButtonKind.js

```diff
@@ -0,0 +1,221 @@
+import styled, { css } from 'styled-components';
+
+import {
+  backgroundStyle,
+  focusStyle,
+  genericStyles,
+  normalizeColor,
+} from '../../utils';
+import { defaultProps } from '../../default-props';
+
+const radiusStyle = props => {
+  const size = props.sizeProp;
+  if (size && props.theme.button.size && props.theme.button.size[size]) {
+    return props.theme.button.size[size].border.radius;
+  }
+  return props.theme.button.border.radius;
+};
+
+const fontStyle = props => {
+  const size = props.sizeProp || 'medium';
+  const data = props.theme.text[size];
+  return css`
+    font-size: ${data.size};
+    line-height: ${data.height};
+  `;
+};
+
+const padStyle = ({ sizeProp: size, theme }) => {
+  if (size && theme.button.size && theme.button.size[size]) {
+    return css`
+      padding: ${theme.button.size[size].padding.vertical}
+        ${theme.button.size[size].padding.horizontal};
+    `;
+  }
+
+  if (theme.button.padding) {
+    return css`
+      padding: ${theme.global.edgeSize[theme.button.padding.vertical] ||
+          theme.button.padding.vertical}
+        ${theme.global.edgeSize[theme.button.padding.horizontal] ||
+          theme.button.padding.horizontal};
+    `;
+  }
+  return '';
+};
+
+// The > svg rule is to ensure Buttons with just an icon don't add additional
+// vertical height internally.
+const basicStyle = props => css`
+  border: none;
+  border-radius: ${radiusStyle(props)};
+  ${padStyle(props)}
+  ${fontStyle(props)}
+
+  > svg {
+    vertical-align: bottom;
+  }
+`;
+
+// CSS for this sub-object in the theme
+const kindPartStyles = (obj, theme, colorValue) => {
+  const styles = [];
+  if (obj.padding) {
+    if (obj.padding.vertical)
+      styles.push(
+        `padding: ${theme.global.edgeSize[obj.padding.vertical] ||
+          obj.padding.vertical ||
+          0} ${theme.global.edgeSize[obj.padding.horizontal] ||
+          obj.padding.horizontal ||
+          0};`,
```

Comment:
I didn't get a chance to test it yet, but where does the parsing of `horizontal` and `vertical` (to `left`, `top` etc) is happening? should this logic be using `parseMetricToNum`?

Reply:
The CSS `padding: a b;` means top and bottom get "a" and left and right get "b". We don't need `parseMetricToNum()` because edgeSize wants a string, not a number, and the CSS will take whatever is passed through.

Reply:
Thanks for clarifying.